homeassistant/components/tuya/switch.py

```diff
@@ -475,6 +475,22 @@
             entity_category=EntityCategory.CONFIG,
         ),
     ),
+    # Smart Gardening system
+    # https://developer.tuya.com/en/docs/iot/categorysz?id=Kaiuz4e6h7up0
+    "sz": (
+        SwitchEntityDescription(
+            key=DPCode.SWITCH,
+            translation_key="power",
+        ),
+        SwitchEntityDescription(
+            key=DPCode.PUMP,
+            translation_key="pump",
+        ),
+        SwitchEntityDescription(
+            key=DPCode.LIGHT,
+            translation_key="light_spectrum_veg_herb",
+        ),
```

Comment:
This is a light, according to specifications, and thus should be a light entity.

![CleanShot 2024-05-29 at 08 41 28@2x](https://github.com/home-assistant/core/assets/195327/85d68112-524d-41ba-a81b-237e9f4cd6d0)

Reply:
Updated